pkg/cluster/mpi/istiopeerauthentication.go

```diff
@@ -0,0 +1,40 @@
+package mpi
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dcv1alpha1 "github.com/dominodatalab/distributed-compute-operator/api/v1alpha1"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/cluster/metadata"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/components"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/controller/core"
+	"github.com/dominodatalab/distributed-compute-operator/pkg/resources/istio"
+)
+
+func IstioPeerAuthentication(enabled bool) core.Component {
```

Comment:
Should it return `OwnedComponent`?

Reply:
This seems to be tricky because the object being returned is imported from the Istio library, and so is missing some methods I think (deepcopy?)